src/app/libp2p_helper/src/libp2p_helper/stream_msg.go

```diff
@@ -133,15 +121,8 @@ func (m OpenStreamReq) handle(app *app, seqno uint64) *capnp.Message {
 		return mkRpcRespError(seqno, badp2p(err))
 	}
 
-	app.StreamsMutex.Lock()
-	defer app.StreamsMutex.Unlock()
-	app.Streams[streamIdx] = stream
-	go func() {
-		// FIXME HACK: allow time for the openStreamResult to get printed before we start inserting stream events
```

Comment:
I need confirmation but I think this is a leftover from an older implementation that didn't track sequence numbers and is not needed anymore?

Reply:
This hack might still be needed.

The code tries to synchronize calls to `app.writeMsg`: call corresponding to `openStream` response should precede the call about message received. 

Is the change necessary?

I know how we could do it the right way, but it would require some effort.

P.S. seems like the ordering on `openStream` and messages within might not be necessary, as it seems like the first message on stream always comes from the stream initiator. But I'd prefer to avoid relying on this much, hence if we are to remove the hack, I'd preserve ordering in a different way.

Reply:
Ok, I will restore it

Reply:
I removed this hack in commit https://github.com/MinaProtocol/mina/pull/14467/commits/036464d98d2d12116830e490c6cb00d4a9d5eda3